snappy_wrappers/wrappers/pvactools/pvacseq/wrapper.py

```diff
@@ -0,0 +1,159 @@
+# Replace with full absolute paths
+files_to_bind = {k: os.path.realpath(v) for k, v in files_to_bind.items()}
+# Directories that mut be bound
+dirs_to_bind = {k: os.path.dirname(v) for k, v in files_to_bind.items()}
+# List of unique directories to bind: on cluster: <directory> -> from container: /bindings/d<i>)
+bound_dirs = {e[1]: e[0] for e in enumerate(list(set(dirs_to_bind.values())))}
+# Binding command
+bindings = " ".join(["-B {}:/bindings/d{}".format(k, v) for k, v in bound_dirs.items()])
+bound_files = {
+    k: "/bindings/d{}/{}".format(bound_dirs[dirs_to_bind[k]], os.path.basename(v))
+    for k, v in files_to_bind.items()
+}
+
+shell.executable("/bin/bash")
+
+shell(
+    r"""
+# -----------------------------------------------------------------------------
+# Redirect stderr to log file by default and enable printing executed commands
+# Also pipe stderr to log file
+if [[ -n "{snakemake.log.log}" ]]; then
+    if [[ "$(set +e; tty; set -e)" != "" ]]; then
+        rm -f "{snakemake.log.log}" && mkdir -p $(dirname {snakemake.log.log})
+        exec 2> >(tee -a "{snakemake.log.log}" >&2)
+    else
+        rm -f "{snakemake.log.log}" && mkdir -p $(dirname {snakemake.log.log})
+        echo "No tty, logging disabled" >"{snakemake.log.log}"
+    fi
+fi
+# Write out information about conda installation.
+conda list >{snakemake.log.conda_list}
+conda info >{snakemake.log.conda_info}
+md5sum {snakemake.log.conda_list} >{snakemake.log.conda_list_md5}
+md5sum {snakemake.log.conda_info} >{snakemake.log.conda_info_md5}
+
+# Setup auto-cleaned tmpdir
+export TMPDIR=$(mktemp -d)
+trap "rm -rf $TMPDIR" EXIT
+
+# Compute md5 checksum
+md5() {{
+    fn=$1
+    d=$(dirname $fn)
+    f=$(basename $fn)
+    pushd $d 1> /dev/null 2>&1
+    checksum=$(md5sum $f)
+    popd 1> /dev/null 2>&1
+    echo "$checksum"
+}}
+
+set -x
+# -----------------------------------------------------------------------------
+
+# Write out information about conda installation
+conda list > {snakemake.log.conda_list}
+conda info > {snakemake.log.conda_info}
+
+hla_types=$(cat {snakemake.input.hla_normal_dna} {snakemake.input.hla_tumor_dna} {snakemake.input.hla_tumor_rna} | sed 's/^/HLA-/' | sort | uniq | tr '\n' ',' | sed 's/,$//')
+cmd="pvacseq run --normal-sample-name {normal_smaple} \
+    -e1 {epitope_length} \
+    --iedb-install-directory /opt/iedb \
+    -t {snakemake.threads} \
+    {bound_files[combine_vcf]} \
+    {snakemake.wildcards.tumor_library} \
+    $hla_types \
+    {algorithms} \
+    {bound_files[op_dir]} \
+    {BINDING_THRESHOLD}\
+    {percentile_threshold}\
+    {allele_specific_binding_thresholds}\
+    {aggregate_inclusion_binding_threshold}\
+    {NET_CHOP_THRESHOLD}\
+    {PROBLEMATIC_AMINO_ACIDS}\
+    {FAST_SIZE}\
+    {NORMAL_COV}\
+    {TDNA_COV}\
+    {TRNA_COV}\
+    {NORMAL_VAF}\
+    {exclude_NAs}\
+    {maximum_transcript_support_level}"
+echo 'TMPDIR=/bindings/d2' > $TMPDIR/{snakemake.wildcards.tumor_library}.sh
```

Comment:
Why the command is written to a temporary script? Isn't it possible to pass it as a string?